xarray/core/rolling.py

```diff
@@ -267,12 +267,15 @@ def __init__(
         # TODO legacy attribute
         self.window_labels = self.obj[self.dim[0]]
 
-    def __iter__(self) -> Iterator[tuple[RollingKey, DataArray]]:
+    def __iter__(self) -> Iterator[tuple[DataArray, DataArray]]:
         if self.ndim > 1:
             raise ValueError("__iter__ is only supported for 1d-rolling")
-        stops = np.arange(1, len(self.window_labels) + 1)
+
+        offset = int(self.window[0] + 1) // 2 if self.center[0] else 1
+        stops = np.arange(offset, self.obj.sizes[self.dim[0]] + offset)
         starts = stops - int(self.window[0])
-        starts[: int(self.window[0])] = 0
+        starts[: int(self.window[0]) - offset] = 0
+
```

Comment:
Can do `window_0 = int(self.window[0])` and remove all of the `int()` conversions I believe.